### ISSUE TITLE
security: prevent Secrets leaking to debug-level logs when entering device privileged mode

### ISSUE THREAD

**Issue by askbow** [ktbyers/netmiko#2597]
### TL;DR: `secret` field is leaked into logging output when level is `logging.DEBUG` or lower

Netmiko logs all channel communication at Debug level via the Python logging facility, and optionally to a file via its own `SessionLog`.
While `SessionLog` filters the secrets and passwords out, the same is not true for the logger.

When a consumer enables debug-level logging at the root level (for tracing or other purposes), this results in inadvertent commitment of secrets and passwords into the logging system.

## Reproduction

```python
import netmiko
import logging
from getpass import getpass

logging.basicConfig(level=logging.DEBUG, format='%(name)6d %(levelname)s %(message)s')

password = getpass()
with ConnectHandler(
    device_type="cisco_ios",
    host="cisco1.askbow.net",
    username="cisco_test",
    password=password,
    secret=password,
) as nc:
    nc.enable()
```

The logging output will contain raw value of the secret in a `write_channel` message.

### Expected outcome
The logging output SHOULD NOT contain the value of the secret

## Workaround
Consumers who need to capture Debug-level logs from all modules can change logging level for Netmiko for this operation to prevent the secret leaking. This results in minimal loss of debug data.

For example:
```python
with ConnectHandler(
    device_type="cisco_ios",
    host="cisco1.askbow.net",
    username="cisco_test",
    password=password,
    secret=password,
) as nc:
    logging.debug('Set netmiko logging to WARNING to prevent secret leaking')
    nc_logger = logging.getLogger('netmiko')
    base_level = nc_logger.level
    nc_logger.setLevel(logging.WARNING)
    nc.enable()
    nc_logger.setLevel(base_level)
    logging.debug('Restored netmiko logging level')
    # ...
```
